src/rvsdg/egglog_optimizer/ivt.egg

```diff
@@ -0,0 +1,115 @@
+       (let outer-else-block (BaseChangeVecOperand outputs gamma inner-else-block))
+       (let inner-then-block (OperandGroup (SubstVecOperandAll (vec-get gamma-out 1) gamma-in)))
+       (let inner-theta (Theta (BaseChangeOperand pred gamma inner-then-block) 
+                               (PassThroughArguments (vec-length inputs))
+                               (BaseChangeVecOperand outputs gamma inner-then-block)))
+        (union loop 
+           (Gamma (SubstOperandAll pred (VO inputs)) 
+                  (VO inputs) 
+                  (VVO 
+                   (vec-of 
+                       outer-else-block
+                       (BodyToVecOperand (vec-length inputs) inner-theta)))))))
+
+(rewrite (BaseChangeExpr (Const ty ops lit) from to)
+  (Const ty ops lit) :ruleset subst)
+(rewrite (BaseChangeExpr (Call ty f args n-outs) from to)
+  (Call ty f (BaseChangeVecOperand args from to) n-outs) :ruleset subst)
+(rewrite (BaseChangeExpr (badd ty a b) from to) 
+  (badd ty (BaseChangeOperand a from to) (BaseChangeOperand b from to)) :ruleset subst)
+(rewrite (BaseChangeExpr (bsub ty a b) from to) 
+  (bsub ty (BaseChangeOperand a from to) (BaseChangeOperand b from to)) :ruleset subst)
+(rewrite (BaseChangeExpr (bmul ty a b) from to) 
+  (bmul ty (BaseChangeOperand a from to) (BaseChangeOperand b from to)) :ruleset subst)
+(rewrite (BaseChangeExpr (bdiv ty a b) from to) 
+  (bdiv ty (BaseChangeOperand a from to) (BaseChangeOperand b from to)) :ruleset subst)
+(rewrite (BaseChangeExpr (beq ty a b) from to)
+  (beq ty (BaseChangeOperand a from to) (BaseChangeOperand b from to)) :ruleset subst)
+(rewrite (BaseChangeExpr (blt ty a b) from to)
+  (blt ty (BaseChangeOperand a from to) (BaseChangeOperand b from to)) :ruleset subst)
+(rewrite (BaseChangeExpr (bgt ty a b) from to)
+  (bgt ty (BaseChangeOperand a from to) (BaseChangeOperand b from to)) :ruleset subst)
+(rewrite (BaseChangeExpr (ble ty a b) from to)
+  (ble ty (BaseChangeOperand a from to) (BaseChangeOperand b from to)) :ruleset subst)
+(rewrite (BaseChangeExpr (bge ty a b) from to)
+  (bge ty (BaseChangeOperand a from to) (BaseChangeOperand b from to)) :ruleset subst)
+(rewrite (BaseChangeExpr (bnot ty a) from to)
+  (bnot ty (BaseChangeOperand a from to)) :ruleset subst)
+(rewrite (BaseChangeExpr (band ty a b) from to)
+ (band ty (BaseChangeOperand a from to) (BaseChangeOperand b from to)) :ruleset subst)
+(rewrite (BaseChangeExpr (bor ty a b) from to)
+  (bor ty (BaseChangeOperand a from to) (BaseChangeOperand b from to)) :ruleset subst)
+(rewrite (BaseChangeExpr (PRINT a b) from to)
+  (PRINT (BaseChangeOperand a from to) (BaseChangeOperand b from to)) :ruleset subst)
+
+(rewrite (BaseChangeOperand (Arg x) from to) (Arg x) :ruleset subst)
+(rewrite (BaseChangeOperand (Node from) from to) (Node to) :ruleset subst)
+(rewrite (BaseChangeOperand (Project i from) from to) (Project i to) :ruleset subst)
```

Comment:
Is this sound?
Consider the following case:
`(Project i body-a)`
`(Project i body-b)`
`(let basechange (BaseChangeOperand (Project i body-a) body-a body-c))`
Now we use your rewrite, we get:
`(union basechange (Project i body-c))`